brainscore_language/model_helpers/huggingface.py

```diff
@@ -107,10 +109,28 @@ def digest_text(self, text: Union[str, List[str]]) -> Dict[str, DataAssembly]:
                 output['neural'].append(representations)
 
         # merge over text parts
-        output['behavior'] = merge_data_arrays(output['behavior']).sortby('part_number') if output['behavior'] else None
-        output['neural'] = merge_data_arrays(output['neural']).sortby('part_number') if output['neural'] else None
+        self._logger.debug("Merging outputs")
+        output['behavior'] = xr.concat(output['behavior'], dim='presentation').sortby('part_number') \
+            if output['behavior'] else None
+        output['neural'] = xr.concat(output['neural'], dim='presentation').sortby('part_number') \
+            if output['neural'] else None
         return output
 
+    def _tokenize(self, context, num_previous_context_tokens):
+        """
+        Tokenizes the context, keeping track of the newly added tokens in `self.current_tokens`
+        """
+        context_tokens = self.tokenizer(context, truncation=True, return_tensors="pt")
+        context_tokens.to('cuda' if torch.cuda.is_available() else 'cpu')
+        # keep track of tokens in current `text_part`
+        overflowing_encoding: list = np.array(context_tokens.encodings).item().overflowing
+        num_overflowing = 0 if not overflowing_encoding else sum(len(overflow) for overflow in overflowing_encoding)
+        self.current_tokens = {key: value[..., num_previous_context_tokens - num_overflowing:]
+                               for key, value in context_tokens.items()}
+        assert all(torch.numel(value) > 0 for value in self.current_tokens.values()), "lost track of context tokens"
```

Comment:
"lost track"? Isn't it more that the tokens are not relevant to the part of text that we are currently interested in obtaining output for?

Reply:
this assertion is to make sure that the case before these fixes doesn't occur again, i.e. where `self.current_tokens` is just an empty list. I believe this should never trigger, but added it to be sure. If it's confusing, should we remove? We now have tests that cover this too

Reply:
Yes, I think it is quite confusing, so maybe get rid off?

Reply:
done